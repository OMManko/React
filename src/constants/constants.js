export const endPoint = 'http://react-cdp-api.herokuapp.com/movies';

export const moviePosterSizes = {
    SMALL: 'moviePoster__small',
    MEDIUM: 'moviePoster__medium'
};

export const sortOptions = {
    RELEASE: 'release',
    RATING: 'vote_average'
};

export const filterOptions = {
    TITLE: 'title',
    GENRE: 'genres'
};


export const actionTypes = {
    SEARCH_BY: 'SEARCH_BY',
    SORT_BY: 'SORT_BY',
    SEARCH_MOVIES: 'SEARCH_MOVIES',
    UPDATE_INPUT_VALUE: 'UPDATE_INPUT_VALUE',
    SET_SELECTED_MOVIE: 'SET_SELECTED_MOVIE',
    RESET_SELECTED_MOVIE: 'RESET_SELECTED_MOVIE'
};

export const initialState = {
    moviesList: [],
    moviesFound: 0,
    searchInputValue: '',
    selectedSortOption: sortOptions.RELEASE,
    selectedFilterOption: filterOptions.TITLE,
    selectedMovieInfo: {}
};
