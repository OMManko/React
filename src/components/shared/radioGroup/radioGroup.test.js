import { mount, shallow } from 'enzyme';
import React from 'react';
import RadioGroup from './radioGroup';
import { filterOptions } from '../../../constants/constants';

const radiosMock = [
  {
    id: 'title',
    label: filterOptions.TITLE,
    defaultChecked: true,
  },
  {
    id: 'genre',
    label: filterOptions.GENRE,
  },
];

describe('<RadioGroup />', () => {
  it('should render radio group and match snapshot', () => {
    const component = shallow(<RadioGroup radios={radiosMock}/>);
    expect(component).toMatchSnapshot();
  });

  it('should set title to radio group', () => {
    const title = 'test';
    const component = mount(<RadioGroup title={title} radios={radiosMock}/>);
    expect(component).toMatchSnapshot();
    expect(component.find('.formRadio__groupTitle').text()).toMatch(title);
  });

  it('should set name to radio input', () => {
    const name = 'test';
    const component = mount(<RadioGroup name={name} radios={radiosMock}/>);
    expect(component).toMatchSnapshot();
    expect(component.find('.formRadio').at(0).find('input').props().name).toEqual(name);
  });
});
