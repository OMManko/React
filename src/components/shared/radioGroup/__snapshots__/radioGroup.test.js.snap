// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> should render radio group and match snapshot 1`] = `ShallowWrapper {}`;

exports[`<RadioGroup /> should set name to radio input 1`] = `ReactWrapper {}`;

exports[`<RadioGroup /> should set title to radio group 1`] = `ReactWrapper {}`;
